sunpy/util/decorators.py

```diff
@@ -66,204 +61,17 @@ def new_function():
-                obj_type_name = 'function'
-            elif inspect.ismethod(obj) or isinstance(obj, method_types):
-                obj_type_name = 'method'
-            else:
-                obj_type_name = 'object'
-        else:
-            obj_type_name = obj_type
-
-        if not name:
-            name = get_function(obj).__name__
-
-        altmessage = ''
-        if not message or isinstance(message, type(deprecate)):
-            if pending:
-                message = ('The {func} {obj_type} will be deprecated in '
-                           'version {deprecated_version}.')
-            else:
-                message = ('The {func} {obj_type} is deprecated and may '
-                           'be removed in {future_version}.')
-            if alternative:
-                altmessage = f'\n        Use {alternative} instead.'
-
-        if removal_version is None:
-            future_version = 'a future version'
-        else:
-            future_version = f'version {removal_version}'
-
-        message = ((message.format(**{
-            'func': name,
-            'name': name,
-            'deprecated_version': since,
-            'future_version': future_version,
-            'alternative': alternative,
-            'obj_type': obj_type_name})) +
-            altmessage)
-
-        if isinstance(obj, type):
-            return deprecate_class(obj, message, warning_type)
-        else:
-            return deprecate_function(obj, message, warning_type)
-
-    if isinstance(message, type(deprecate)):
-        return deprecate(message)
-
-    return deprecate
-
-
-class add_common_docstring:
-    """
-    A function decorator that will append and/or prepend an addendum to the
-    docstring of the target function.
-
-    Parameters
-    ----------
-    append : `str`, optional
-        A string to append to the end of the functions docstring.
-
-    prepend : `str`, optional
-        A string to prepend to the start of the functions docstring.
-
-    **kwargs : `dict`, optional
-        A dictionary to format append and prepend strings.
-    """
-
-    def __init__(self, append=None, prepend=None, **kwargs):
-        if kwargs:
-            append = append
-            prepend = prepend
-        self.append = append
-        self.prepend = prepend
-        self.kwargs = kwargs
-
-    def __call__(self, func):
-        func.__doc__ = func.__doc__ if func.__doc__ else ''
-        self.append = self.append if self.append else ''
-        self.prepend = self.prepend if self.prepend else ''
-        if self.append and isinstance(func.__doc__, str):
-            func.__doc__ += self.append
-        if self.prepend and isinstance(func.__doc__, str):
-            func.__doc__ = self.prepend + func.__doc__
-        if self.kwargs:
-            func.__doc__ = func.__doc__.format(**self.kwargs)
-        return func
+    return _deprecated(
+        since=since, message=message, name=name, alternative=alternative, pending=False,
```

Comment:
We do not support pending as that can not have its warning replaced, we could subclass the class and fix it?